lib/Renard/Curie/Data/PDF.pm

```diff
@@ -2,43 +2,101 @@ use Modern::Perl;
 package Renard::Curie::Data::PDF;
 
 use Capture::Tiny qw(capture_stdout);
+use XML::Simple;
+use Path::Tiny;
+use Alien::MuPDF 0.004;
 
-sub mudraw_get_pdf_page_as_png {
-	my ($pdf_filename, $pdf_page_no) = @_;
+BEGIN {
+	our $MUTOOL_PATH =  path(Alien::MuPDF->dist_dir, qw(bin mutool));
+}
 
+sub _call_mutool {
+	my @args = ( $Renard::Curie::Data::PDF::MUTOOL_PATH, @_ );
 	my ($stdout, $exit) = capture_stdout {
-		system("mudraw",
-			qw( -F png ),
-			qw( -o -),
-			$pdf_filename,
-			$pdf_page_no,
-		);
+		system( @args );
 	};
 
-	die "Unexpected mudraw exit: $exit" if $exit;
+	die "Unexpected mutool exit: $exit" if $exit;
+
+	return $stdout;
+}
+
+sub get_mutool_pdf_page_as_png {
+	my ($pdf_filename, $pdf_page_no) = @_;
+
+	my $stdout = _call_mutool(
+		qw(draw),
+		qw( -F png ),
+		qw( -o -),
+		$pdf_filename,
+		$pdf_page_no,
+	);
+
+	return $stdout;
+}
+
+sub get_mutool_text_stext_raw {
+	my ($pdf_filename, $pdf_page_no) = @_;
+
+	my $stdout = _call_mutool(
+		qw(draw),
+		qw(-F stext),
+		qw(-o -),
+		$pdf_filename,
+		$pdf_page_no,
+	);
 
 	return $stdout;
 }
 
-sub get_pdfinfo_for_filename {
+sub get_mutool_text_stext_xml {
+	my ($pdf_filename, $pdf_page_no) = @_;
+
+	my $stext_xml = get_mutool_text_stext_raw(
+		$pdf_filename,
+		$pdf_page_no,
+	);
+	# page -> [list of blocks]
+	#   block -> [list of blocks]
+	#     block is either:
+	#       - stext
+	#           line -> [list of lines] (all have same baseline)
+	#             span -> [list of spans] (horizontal spaces over a line)
+	#               char -> [list of chars]
+	#       - image
+	#           TODO
+
+	my $stext = XMLin( $stext_xml,
+		ForceArray => [ qw(page block line span char) ] );
+
+	return $stext;
+}
+
+sub get_mutool_page_info_raw {
```

Comment:
This calls `mutool pages` and gets the page bounds (e.g., CropBox) in an XML format.